src/Umbraco.Forms.Integrations.Crm.Hubspot/Services/HubspotContactService.cs

```diff
@@ -160,7 +160,9 @@ public async Task<CommandResult> PostContactAsync(Record record, List<MappedProp
                 var recordField = record.GetRecordField(Guid.Parse(fieldId));
                 if (recordField != null)
                 {
-                    var value = recordField.ValuesAsHubspotString(false);
+                    var value = _settings.AllowContactUpdate && mapping.AppendValue
```

Comment:
That's nice if we don't have to do the extra step of retrieving the existing record.

But might be worth checking 100% that this works, and you don't end up with duplicates.  We'd want to store full set of unique values stored in the current form submission and the existing HubSpot record.

Maybe HubSpot handles that on their end, but I'd suggest testing each of these scenarios:

- Submit new with no, one and two selections.
- Submit update with none, one and two new selections.
- Submit update with one or two new selections that include selections already made.

Reply:
Yes, that sounds like the behaviour the person that requested this is expecting.